lib/internal/bootstrap/node.js

```diff
@@ -174,8 +184,24 @@ if (config.hasInspector) {
 // This attaches some internal event listeners and creates:
 // process.send(), process.channel, process.connected,
 // process.disconnect()
-if (isMainThread && process.env.NODE_CHANNEL_FD) {
-  mainThreadSetup.setupChildProcessIpcChannel();
+if (process.env.NODE_CHANNEL_FD) {
+  if (isMainThread) {
```

Comment:
I think it makes sense for this to be `ownsProcessState` (which is available when rebasing against master)?

Reply:
Updated!